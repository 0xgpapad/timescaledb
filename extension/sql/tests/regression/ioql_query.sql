\set ON_ERROR_STOP 1

\o /dev/null
\ir include/insert.sql

\o
\set ECHO ALL
\c Test1

SELECT *
FROM ioql_exec_query(new_ioql_query(hypertable_name => 'testNs'));

\set ON_ERROR_STOP 0
SELECT *
FROM ioql_exec_query(new_ioql_query(hypertable_name => 'DoesNotExist'));
\set ON_ERROR_STOP 1

SELECT *
FROM ioql_exec_query(new_ioql_query(hypertable_name => 'testNs',
                                    select_items => ARRAY [new_select_item('series_0', 'SUM')],
                                    aggregate => new_aggregate((100 * 60 * 60 * 1e9) :: BIGINT)
                     ));

SELECT *
FROM ioql_exec_query(new_ioql_query(hypertable_name => 'testNs',
                                    select_items => ARRAY [new_select_item('series_0', 'SUM')],
                                    aggregate => new_aggregate((100 * 60 * 60 * 1e9) :: BIGINT),
                                    limit_rows => 1
                     ));

SELECT *
FROM ioql_exec_query(new_ioql_query(hypertable_name => 'testNs',
                                    select_items => ARRAY [new_select_item('series_0', 'SUM')],
                                    aggregate => new_aggregate((100 * 60 * 60 * 1e9) :: BIGINT, 'series_1')
                     ));

SELECT *
FROM ioql_exec_query(new_ioql_query(hypertable_name => 'testNs',
                                    select_items => ARRAY [new_select_item('series_0', 'SUM')],
                                    aggregate => new_aggregate((100 * 60 * 60 * 1e9) :: BIGINT, 'series_1'),
                                    limit_rows => 1
                     ));

SELECT *
FROM ioql_exec_query(new_ioql_query(hypertable_name => 'testNs',
                                    select_items => ARRAY [new_select_item('series_0', 'SUM')],
                                    aggregate => new_aggregate((100 * 60 * 60 * 1e9) :: BIGINT, 'series_1'),
                                    limit_time_periods => 1
                     ));

SELECT *
FROM ioql_exec_query(new_ioql_query(hypertable_name => 'testNs',
                                    select_items => ARRAY [new_select_item('series_0')],
                                    limit_by_column => new_limit_by_column('device_id', 1)
                     ));

SELECT *
FROM ioql_exec_query(new_ioql_query(hypertable_name => 'testNs',
                                    select_items => ARRAY [new_select_item('series_0')],
                                    limit_rows => 1
                     ));
