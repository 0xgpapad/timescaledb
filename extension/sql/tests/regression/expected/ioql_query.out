\c Test1
SELECT *
FROM ioql_exec_query(new_ioql_query(namespace_name => 'testNs'));
                                                         json                                                          
-----------------------------------------------------------------------------------------------------------------------
 {"timeCustom":1257987600000000000,"device_id":"dev1","series_0":1.5,"series_1":1,"series_2":null,"series_bool":null}
 {"timeCustom":1257987600000000000,"device_id":"dev1","series_0":1.5,"series_1":2,"series_2":null,"series_bool":null}
 {"timeCustom":1257897600000000000,"device_id":"dev1","series_0":4.5,"series_1":5,"series_2":null,"series_bool":false}
 {"timeCustom":1257894002000000000,"device_id":"dev1","series_0":2.5,"series_1":3,"series_2":null,"series_bool":null}
 {"timeCustom":1257894001000000000,"device_id":"dev1","series_0":3.5,"series_1":4,"series_2":null,"series_bool":null}
 {"timeCustom":1257894000000001000,"device_id":"dev1","series_0":2.5,"series_1":3,"series_2":null,"series_bool":null}
 {"timeCustom":1257894000000000000,"device_id":"dev1","series_0":1.5,"series_1":1,"series_2":2,"series_bool":true}
 {"timeCustom":1257894000000000000,"device_id":"dev1","series_0":1.5,"series_1":2,"series_2":null,"series_bool":null}
 {"timeCustom":1257894000000000000,"device_id":"dev20","series_0":1.5,"series_1":1,"series_2":null,"series_bool":null}
 {"timeCustom":1257894000000000000,"device_id":"dev20","series_0":1.5,"series_1":2,"series_2":null,"series_bool":null}
(10 rows)

\set ON_ERROR_STOP 0
SELECT *
FROM ioql_exec_query(new_ioql_query(namespace_name => 'DoesNotExist'));
\set ON_ERROR_STOP 1
SELECT *
FROM ioql_exec_query(new_ioql_query(namespace_name => 'testNs',
                                    select_items => ARRAY [new_select_item('series_0', 'SUM')],
                                    aggregate => new_aggregate((100 * 60 * 60 * 1e9) :: BIGINT)
                     ));
                      json                       
-------------------------------------------------
 {"time":1257840000000000000,"sum(series_0)":22}
(1 row)

SELECT *
FROM ioql_exec_query(new_ioql_query(namespace_name => 'testNs',
                                    select_items => ARRAY [new_select_item('series_0', 'SUM')],
                                    aggregate => new_aggregate((100 * 60 * 60 * 1e9) :: BIGINT),
                                    limit_rows => 1
                     ));
                      json                       
-------------------------------------------------
 {"time":1257840000000000000,"sum(series_0)":22}
(1 row)

SELECT *
FROM ioql_exec_query(new_ioql_query(namespace_name => 'testNs',
                                    select_items => ARRAY [new_select_item('series_0', 'SUM')],
                                    aggregate => new_aggregate((100 * 60 * 60 * 1e9) :: BIGINT, 'series_1')
                     ));
                             json                              
---------------------------------------------------------------
 {"series_1":3,"time":1257840000000000000,"sum(series_0)":5}
 {"series_1":2,"time":1257840000000000000,"sum(series_0)":4.5}
 {"series_1":4,"time":1257840000000000000,"sum(series_0)":3.5}
 {"series_1":1,"time":1257840000000000000,"sum(series_0)":4.5}
 {"series_1":5,"time":1257840000000000000,"sum(series_0)":4.5}
(5 rows)

SELECT *
FROM ioql_exec_query(new_ioql_query(namespace_name => 'testNs',
                                    select_items => ARRAY [new_select_item('series_0', 'SUM')],
                                    aggregate => new_aggregate((100 * 60 * 60 * 1e9) :: BIGINT, 'series_1'),
                                    limit_rows => 1
                     ));
                             json                              
---------------------------------------------------------------
 {"series_1":1,"time":1257840000000000000,"sum(series_0)":4.5}
(1 row)

SELECT *
FROM ioql_exec_query(new_ioql_query(namespace_name => 'testNs',
                                    select_items => ARRAY [new_select_item('series_0', 'SUM')],
                                    aggregate => new_aggregate((100 * 60 * 60 * 1e9) :: BIGINT, 'series_1'),
                                    limit_time_periods => 1
                     ));
                             json                              
---------------------------------------------------------------
 {"series_1":3,"time":1257840000000000000,"sum(series_0)":5}
 {"series_1":2,"time":1257840000000000000,"sum(series_0)":4.5}
 {"series_1":4,"time":1257840000000000000,"sum(series_0)":3.5}
 {"series_1":1,"time":1257840000000000000,"sum(series_0)":4.5}
 {"series_1":5,"time":1257840000000000000,"sum(series_0)":4.5}
(5 rows)

SELECT *
FROM ioql_exec_query(new_ioql_query(namespace_name => 'testNs',
                                    select_items => ARRAY [new_select_item('series_0')],
                                    limit_by_field => new_limit_by_field('device_id', 1)
                     ));
                                 json                                  
-----------------------------------------------------------------------
 {"timeCustom":1257987600000000000,"series_0":1.5,"device_id":"dev1"}
 {"timeCustom":1257894000000000000,"series_0":1.5,"device_id":"dev20"}
(2 rows)

SELECT *
FROM ioql_exec_query(new_ioql_query(namespace_name => 'testNs',
                                    select_items => ARRAY [new_select_item('series_0')],
                                    limit_rows => 1
                     ));
                       json                        
---------------------------------------------------
 {"timeCustom":1257987600000000000,"series_0":1.5}
(1 row)

